.github/workflows/self-hosted.yml

```diff
@@ -0,0 +1,50 @@
+name: Run all tests on GPU
+
+on:
+  push:
+  #  branches: ci/*
+  repository_dispatch:
+  schedule:
+    - cron: "0 0 * * *"
+
+jobs:
+  gpu:
+    runs-on: self-hosted
+    steps:
+    - uses: actions/checkout@v2
+    - name: Python version
+      run: |
+        which python
+        python --version
+        pip --version
+    - name: Current dir
+      run: pwd
+    - run: nvidia-smi
+    - name: Create new python env (on self-hosted runners we have to handle isolation ourselves)
+      run: |
+        python -m venv .env
+        source .env/bin/activate
+        which python
+        python --version
+        pip --version
+    - name: Install dependencies
+      run: |
+        source .env/bin/activate
+        pip install .[mecab,sklearn,tf,torch,testing]
+
+    - name: Are GPU recognized by our DL frameworks
+      run: |
+        source .env/bin/activate
+        python -c "import torch; print(torch.cuda.is_available())"
+        python -c "import tensorflow as tf; print(tf.test.is_built_with_cuda()); print(tf.config.list_physical_devices('GPU'))"
```

Comment:
@n1t0  How do you install cudnn? TF says that it can't use our K80 because it cannot find cudnn.  (cc @LysandreJik and @mfuntowicz)

Reply:
I usually follow this: https://docs.nvidia.com/deeplearning/sdk/cudnn-install/index.html#installlinux-tar

Reply:
We installed from the .deb file with @LysandreJik (omg, the Nvidia website is awful)

Reply:
Would it be hard to do the tests with and without CuDNN ? Testing both would cover a broad range of setups.

Reply:
I believe TensorFlow doesn't recognize the available GPUs when CuDNN is not installed?
Are you talking about PyTorch?

Reply:
Note that might be of interest: you can't use the [NVidia docker images](https://docs.nvidia.com/deeplearning/frameworks/pytorch-release-notes/rel_17.12.html#rel_17.12) with the K80 because they require a minimal compute capability of 6.0 whereas the K80 is rated at 3.7.